pkg/stream/producer_unconfirmed.go

```diff
@@ -67,13 +77,14 @@ func (u *unConfirmed) extractWithConfirms(ids []int64) []*ConfirmationStatus {
 			}
 		}
 	}
+	u.maybeUnLock()
 	return res
-
 }
 
 func (u *unConfirmed) extractWithError(id int64, errorCode uint16) *ConfirmationStatus {
 	u.mutexMessageMap.Lock()
 	defer u.mutexMessageMap.Unlock()
+	u.maybeUnLock()
```

Comment:
```suggestion
	cs := u.extract(id, errorCode, false)
	u.maybeUnLock()
	return cs
```
Otherwise, it will be ineffective.